sql: TestScatterWithOneVoter failed

sql.TestScatterWithOneVoter [failed](https://mesolite.cluster.engflow.com/invocations/default/561c5e75-b9f3-44da-a52a-796275b4a7c6?testReportRun=4&testReportShard=6&testReportAttempt=1#targets-Ly9wa2cvc3FsOnNxbF90ZXN0) on release-25.3.6-rc @ [e85eeea015cec27a716de792c527dd83cabcf0ab](https://github.com/cockroachdb/cockroach/commits/e85eeea015cec27a716de792c527dd83cabcf0ab):


```
=== RUN   TestScatterWithOneVoter
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestScatterWithOneVoter16978145
    test_log_scope.go:76: use -show-logs to present logs inline
    test_server_shim.go:143: cluster virtualization disabled in global scope due to issue: #76378 (expected label: C-bug)
    scatter_test.go:270: 
        	Error Trace:	pkg/sql/scatter_test.go:270
        	Error:      	Should not be: 50
        	Test:       	TestScatterWithOneVoter
    panic.go:635: -- test log scope end --
test logs left over in: outputs.zip/logTestScatterWithOneVoter16978145
--- FAIL: TestScatterWithOneVoter (1.05s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=4</code>
 - <code>shard=6</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestScatterWithOneVoter.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-57336

--- a/pkg/sql/scatter_test.go
+++ b/pkg/sql/scatter_test.go
@@ -249,15 +249,39 @@ func TestScatterWithOneVoter(t *testing.T) {
 	if err != nil {
 		t.Fatal(err)
 	}
+	t.Logf("before scatter: replica distribution across stores: %v (total: %d)",
+		oldReplicaCounts, oldTotalReplicas)
 
 	// Expect that the number of replicas on store 1 to have changed. We can't
 	// assert that the distribution will be even across all three stores, but s1
 	// (the initial leaseholder and replica) should have a different number of
 	// replicas than before. Rebalancing is otherwise disabled in this test, so
 	// the only replica movements are from the scatter.
+
+	// Enable tracing to capture detailed scatter execution information.
+	r.Exec(t, "SET tracing = on")
 	r.Exec(t, "ALTER TABLE test.t SCATTER")
+	r.Exec(t, "SET tracing = off")
+
+	// If the test fails, print the trace of the scatter statement to help
+	// us investigate failures.
+	defer func() {
+		if !t.Failed() {
+			return
+		}
+
+		traceRows := r.QueryStr(t, "SELECT age, message FROM [SHOW TRACE FOR SESSION]")
+		if len(traceRows) > 0 {
+			t.Logf("SCATTER trace (%d relevant messages):", len(traceRows))
+			for _, row := range traceRows {
+				t.Logf("[%s] %s", row[0], row[1])
+			}
+		}
+	}()
+
 	newReplicaCounts, newTotalReplicas, err := getReplicaCounts()
 	require.NoError(t, err)
+	t.Logf("after scatter: replica distribution across stores: %v (total: %d)", newReplicaCounts, newTotalReplicas)
 
 	require.Equal(t, oldTotalReplicas, newTotalReplicas,
 		"expected total replica count to remain the same post-scatter, "+

